src/extension/importer/core/src/main/java/org/geoserver/importer/Importer.java

```diff
@@ -1731,48 +1765,28 @@ public File getImportRoot() {
     }
 
     public File getUploadRoot() {
-        String value = null;
-        try {
-            value = System.getProperty(UPLOAD_ROOT_KEY);
-            if (value == null) {
-                value = System.getenv(UPLOAD_ROOT_KEY);
-            }
-        } catch (Throwable ex) {
-            if (LOGGER.isLoggable(Level.FINEST)) {
-                LOGGER.log(
-                        Level.FINEST,
-                        "Could not access system property '" + UPLOAD_ROOT_KEY + "': " + ex);
-            }
-        }
-
-        if (configFile != null && configFile.isModified()) {
-            try {
-                props = configFile.read();
-            } catch (Exception e) {
-                LOGGER.log(
-                        Level.FINEST,
-                        "Could not find any '" + PROPERTYFILENAME + "' property file.",
-                        e);
-            }
-        }
-
-        if (props != null && value == null) {
-            value = props.getProperty(UPLOAD_ROOT_KEY);
-        }
+        String value = configuration.getUploadRoot();
 
         try {
             if (value != null) {
+                // clean up prefixes added by the file chooser, in the UI
```

Comment:
If this is a common boilerplate for handling values from the file chooser it may be worth making a utility method (for example in Paths)

Reply:
Ah, sorry, I have a change that will fix this part without requiring any cleanup.